src/effects/Saturation.cpp

```diff
@@ -102,6 +109,58 @@ std::shared_ptr<Frame> Saturation::GetFrame(std::shared_ptr<Frame> frame, int64_
 		G = constrain(G);
 		B = constrain(B);
 
+		/*
+		 * Color-separated saturation adjustment
+		 *
+		 * Splitting each of the three subpixels (R, G and B) into three distincs sub-subpixels (R, G and B in turn)
+		 * which in their optical sum reproduce the original subpixel's color OR produce white light in the brightness
+		 * of the original subpixel (dependening on the color channel's slider value).
+		 */
+
+		// Three subpixels producing either R or white with brightness of R
+		int Rr = R;
+		int Gr = 0;
+		int Br = 0;
+
+		// Three subpixels producing either G or white with brightness of G
+		int Rg = 0;
+		int Gg = G;
+		int Bg = 0;
+
+		// Three subpixels producing either B or white with brightness of B
+		int Rb = 0;
+		int Gb = 0;
+		int Bb = B;
+
+		// Compute the brightness ("saturation multiplier") of the replaced subpixels
+		// Actually mathematical no-ops mostly, verbosity is kept just for clarification
+		const double p_r = sqrt( (Rr * Rr * pR) + (Gr * Gr * pG) + (Br * Br * pB) );
+		const double p_g = sqrt( (Rg * Rg * pR) + (Gg * Gg * pG) + (Bg * Bg * pB) );
+		const double p_b = sqrt( (Rb * Rb * pR) + (Gb * Gb * pG) + (Bb * Bb * pB) );
```

Comment:
> Actually mathematical no-ops mostly, verbosity is kept just for clarification

And while that's generally appreciated (certainly, compared to the alternative of completely-impenetrable magic code!), I think it might be excessive here because I'm not sure the extraneous bits of math _actually do_ add clarity — I feel like it's kind of the opposite, they just obscure the relatively-simpler math that's actually happening.

There's no need for `Rr`, `Gg`, and `Bb` since they're just `R`, `G`, and `B`, the rest of the subchannel vars are all `0` and can be eliminated, and the multipliers are really just:
```c++
const double p_r = sqrt( R * R * pR );
const double p_g = sqrt( G * G * pG );
const double p_b = sqrt( B * B * pB );
```
And so the adjustments just become:
```c++
R = p_r + (R - p_r) * saturation_value_R;
G = p_g + (G - p_g) * saturation_value_G;
B = p_b + (B - p_b) * saturation_value_B;
```
And done. Do I have that right? If so, I'd just write it that way (but with explanatory comments, still); it doesn't need to look so much more complicated.

Reply:
+1

BTW, that's correct, but it is exactly what I meant when I recently proposed to manually cancel the variables, but you answered, this is not needed as the compiler's optimizer would do that. ;-)

Reply:
Well, I did say "most likely"... and while it's true the `0` math will optimize away, and _possibly_ the variables that only ever store `0` will as well, things like copying `R` to `Rr`, just to eventually copy it back to `R` after doing some math on it, the compiler won't be able to completely eliminate.

It's a measly few bytes, of course, and certainly not any kind of realistic performance or resource-consumption worry... but unnecessary, all the same. Though my original statement stands, I'm not really hung up about it and this is not the hill I want to die on. Just tossing it out there, since you _had_ originally made that offer. :wink:

Reply:
Fixed by c979f5b0d6655add38016ab058d26569cc49361f.